TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/EditorTests/FeatureTests/IncludeTests.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using UnityEditor.ShaderGraph;
+using UnityEngine;
+
+namespace UnityEditor.ShaderFoundry.UnitTests
+{
+    class IncludeTests // : BlockTestRenderer
+    {
+        [Test]
+        public void BlockInclude_ShowsUpInGeneratedShader()
+        {
+            var container = new ShaderContainer();
+
+            var blocks = new List<Block>();
+
+            // We may need to add actual used inputs and outputs to the block, if the linker implements block culling..
+            var blockBuilder = BlockBuilderUtilities.DefineSimpleBlock(container, "MyBlock", container._float4, "inValue", container._float4, "outValue", "inputs.inValue");
+
+            // TODO: we need to make quotes be added by the generator... shouldn't have to add them to the string specified here
+            blockBuilder.AddInclude(new IncludeDescriptor.Builder(container, "\"MyIncludeFile\"").Build());
+            blocks.Add(blockBuilder.Build());
+
+            var shaderCode = BlockTestRenderer.BuildSimpleSurfaceShader(container, "TestShader", blocks);
+            Debug.Log(shaderCode);
+
+            string expectedCode = "#include \"MyIncludeFile\"";
+            Assert.AreNotEqual(-1, shaderCode.IndexOf(expectedCode), $"Expected code missing: {expectedCode}");
+        }
+
+        [Test]
+        public void BlockFunctionInclude_ShowsUpInGeneratedShader()
```

Comment:
I'd try to improve the names a bit more to follow the "given when then" pattern. Something like
"BlockFunctionWithInclude_GeneratingShader_IncludeIsFound"

Reply:
https://internaldocs.unity.com/automation/TestFrameworks/NativeTests/BestPractices/
```
Name tests following the pattern: MethodOrFunctionName_ThisIsWhatIsExpected or MethodOrFunctionName_InputDescription_OutputOrBehaviourDescription or GivenSetup_WhenAction_ThenExpectedResult. Going with scenario-based names, is just as fine: CanAddFileWhileDatabaseIsLocked.
```

I think what you did is close to the first version, it's just not quite descriptive of what you're doing. I'll leave it up to you though.